cpp/src/join/semi_join.cu

```diff
@@ -34,10 +35,26 @@
 #include <thrust/copy.h>
 #include <thrust/distance.h>
 #include <thrust/sequence.h>
+#include <thrust/tuple.h>
+
+#include <cuco/static_map.cuh>
 
 namespace cudf {
 namespace detail {
 
+namespace {
+/**
+ * @brief Device functor to create a pair of hash value and index for a given row.
+ */
+struct make_pair_function {
+  __device__ __forceinline__ cudf::detail::pair_type operator()(size_type i) const noexcept
+  {
+    return cuco::make_pair<hash_value_type, size_type>(i, true);
```

Comment:
I'm a bit confused by the key/value types: this returns a pair of `size_type` and `bool` but the declared types are `hash_value_type` and `size_type`.

Reply:
That is a good point. For the current implementation the value is unused and is therefore irrelevant. That matches the old implementation here, but I can change it to insert an arbitrary numeric value instead for now (with a comment). Once we've implemented a set in cuco we should be able to switch to using that, so this is a stopgap solution until then.

Reply:
Right, thanks for the clarification.